src/Players/AIPlayer.hs

```diff
@@ -87,9 +88,29 @@ getMoveWithMaxScore gameTree = ChessMove ('z', -1) ('z', -1)
 -- The higher the score, the more advantageous the scenario is for the colour
 -- First version: score is the sum of the scores for each piece on the board of the colour minus the scores for each piece of the other colour on the board
 -- Scores per piece: https://cdn-media-1.freecodecamp.org/images/1*e4p9BrCzJUdlqx7KVGW9aA.png
--- TODO: implement + test this [Yiyi]
 score :: ChessBoard -> ChessPieceColour -> Integer
-score chessBoard forColour = -1
+score (ChessBoard pieces _) colour = scoreOfCurrColour - scoreOfOppoColour
+    where currColour = filter (\ (position, piece) -> (getPieceColour piece) == colour) pieces
+          oppoColour = filter (\ (position, piece) -> (getPieceColour piece) /= colour) pieces
```

Comment:
I believe `getPieceColour` returns `Maybe ChessPieceColour` (you'll have to double check) in which case you either have to do `(getPieceColour piece) == (Just colour)` or `(fromJust (getPieceColour piece)) == colour`
```suggestion
    where currColour = filter (\ (position, piece) -> (getPieceColour piece) == (Just colour)) pieces
          oppoColour = filter (\ (position, piece) -> (getPieceColour piece) /= (Just colour)) pieces
```

Reply:
i just double checked, `getPieceColour` returns `ChessPieceColour` (`getPieceColour :: ChessPiece -> ChessPieceColour`)